### PR TITLE
Move notifications about Courtesy push activities from Slack to MS Teams

### PR DESCRIPTION
**Pull request by denis-tikhomirov**
**Task name**: -

**Description**: Move notifications about Courtesy push activities from Slack to MS Teams. Content of notifications was updated.

**Documentation changes required:** (Y/N) No

**Added unit tests:** (Y/N) No

**Attached related issue:** (Y/N) microsoft/build-task-team/issues/2123

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it - Not applicable for the changes
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Review by KonstantinTyukalov — APPROVED** [review]

File: ci/courtesy-push.yml [hunk 1]
```diff
@@ -59,7 +59,13 @@ steps:
...
-    Invoke-RestMethod -Uri $(slackUri) -Method Post -Body $body -ContentType 'application/json'
-  displayName: 'Send Slack notification'
+    $titleText = "Courtesy Bump of Tasks PR created - ID $($env:PrID)"
+    $messageText = "Created Courtesy Bump of Tasks PR. Please review and approve/merge [PR $($env:PrID)]($($env:PrLink)). Related article in [Wiki](https://mseng.visualstudio.com/AzureDevOps/_wiki/wikis/AzureDevOps.wiki/25317/Release-of-pipeline-tasks)."
+    $body = [PSCustomObject]@{
+        title = $titleText
+        text = $messageText
+        themeColor = "#FFFF00"
+      } | ConvertTo-Json
+    Invoke-RestMethod -Uri $(MSTeamsUri) -Method Post -Body $body -ContentType 'application/json'
```
Comment (alexander-smolyakov) [inline ci/courtesy-push.yml:69]: As I can see the code for both notifications are quite similar, in this case, could it make sense to extract the coed to the file and reuse it?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -73,12 +73,17 @@ jobs:
...
-    displayName: 'Send notification to the slack channel'
+      # Notify MS Teams channel about failed courtesy push PR creation
+      $pipelineLink = "$env:SYSTEM_TEAMFOUNDATIONCOLLECTIONURI$env:SYSTEM_TEAMPROJECT/_build/results?buildId=$env:BUILD_BUILDID&_a=summary"
+      $titleText = "Courtesy push build failed - ID $($env:BUILD_BUILDID)"
+      $messageText = "Failed to create Courtesy Bump of Tasks PR. Please review the results of failed build [ID $($env:BUILD_BUILDID)]($($pipelineLink)). Related article in [Wiki](https://mseng.visualstudio.com/AzureDevOps/_wiki/wikis/AzureDevOps.wiki/25317/Release-of-pipeline-tasks)."
+      $body = [PSCustomObject]@{
+        title = $titleText
+        text = $messageText
+        themeColor = "#FF0000"
+      } | ConvertTo-Json
```
Comment (alexander-smolyakov) [inline azure-pipelines.yml:84]: Could you please add some comments or the link to the related docs regarding which properties can be used in the body?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -73,12 +73,17 @@ jobs:
...
-      # Notify slack channel about failed courtesy push PR creation
-      $pipelineLink = "$env:SYSTEM_TEAMFOUNDATIONCOLLECTIONURI$env:SYSTEM_TEAMPROJECT/_build/results?buildId=$env:BUILD_BUILDID&_a=summary" 
-      $messageText = "Failed to create courtesy push PR. Please review the results of failed build: $($pipelineLink) "
-      $body = '{"text": "'+$($messageText)+'"}'
-      Invoke-RestMethod -Uri $(slackUri) -Method Post -Body $body -ContentType 'application/json'
-    displayName: 'Send notification to the slack channel'
+      # Notify MS Teams channel about failed courtesy push PR creation
+      $pipelineLink = "$env:SYSTEM_TEAMFOUNDATIONCOLLECTIONURI$env:SYSTEM_TEAMPROJECT/_build/results?buildId=$env:BUILD_BUILDID&_a=summary"
+      $titleText = "Courtesy push build failed - ID $($env:BUILD_BUILDID)"
+      $messageText = "Failed to create Courtesy Bump of Tasks PR. Please review the results of failed build [ID $($env:BUILD_BUILDID)]($($pipelineLink)). Related article in [Wiki](https://mseng.visualstudio.com/AzureDevOps/_wiki/wikis/AzureDevOps.wiki/25317/Release-of-pipeline-tasks)."
```
Comment (alexander-smolyakov) [inline azure-pipelines.yml:79]:
```suggestion
      $buildLink = "[ID $($env:BUILD_BUILDID)]($($pipelineLink))"
      $wikiLink = "[Wiki](https://mseng.visualstudio.com/AzureDevOps/_wiki/wikis/AzureDevOps.wiki/25317/Release-of-pipeline-tasks)"
      $messageText = "Failed to create Courtesy Bump of Tasks PR. Please review the results of failed build $buildLink. Related article in $wikiLink."
```

**Review by alexander-smolyakov — COMMENTED** [review]
Please take a look at the comments below

File: ci/courtesy-push.yml (hunk 1 above)
Comment (denis-tikhomirov) [reply ci/courtesy-push.yml]: Done. Code for notifications is moved to separate script.

File: azure-pipelines.yml (hunk 2 above)
Comment (denis-tikhomirov) [reply azure-pipelines.yml]: Done. Comments are added.

File: azure-pipelines.yml (hunk 3 above)
Comment (denis-tikhomirov) [reply azure-pipelines.yml]: Applied

**Review by alexander-smolyakov — APPROVED** [review]
LGTM, thanks!

**Review by mmrazik — APPROVED** [review]

**Pull request merged**